### DISCUSSION TITLE
lesson 3 : Got the answer but did not understand

### DISCUSSION THREAD

**Discussion by hellxeth** [Cyfrin/foundry-full-course-cu#526] [Q&A]
I have done the lesson 3 challenge and got the answer, but I did not understand anything like how I got it and what is really happening in the code, I was exploring the read contract and after giving some address I got the answer from the read contract but have not understood what exactly happened and how the contract is linked to the lesson we were taught. And when I tried to solve it again the previous answer has not worked. can anyone explain what is happening in the code?

**Comment by shikhar229169** [comment]
As you can see that the LessonThree contract inherits from LessonThreeHelper.

I would suggest you visit that on your own on etherscan and see what are the variables and methods defined in the LessonThreeHelper.
Now as the LessonThree inherits from LessonThreeHelper, so it will have all the methods and variables inherited inside it (exception - private Functions or private variables are not inherited).

So, please check that out first, if you still feel like you are not able to understand it, feel free to ask your doubt.

**Reply by hellxeth** [reply]
That part I understood, Like LessonThree.sol has access to all the methods and variables because we have imported LessonThreeHelper.sol. But in the read contract I Entered some random address in extra description I got 2  and I chose 3 as new location (but according the getArrayElement method we have to get 3 as location right?) why did I get 2?

**Reply by Mepsss** [reply]
not sure which code you are talking about exactly (would help if you could post it).

But from this comment I can only assume. Did you remember that arrays count starting from 0. So you might have put in 2 because you were looking for the address in the 3rd location (0, 1, 2) 2 beeing the 3rd in the array

**Comment by finessevanes** [comment]
Hey hey, I was able to get the answer by the following:
1. find `location`
```solidity
function getArrayElement(uint256 index) external view returns (bool) {
    return s_booleanArray[index];
}
```
this method is inherited into the `Lesson` contract. By going to [read contract](https://sepolia.etherscan.io/address/0x4A62A8901e6113dEF0AEeEC77E531779cd40da73#readContract), I then just started from 0 and continued calling the function until I got true
    
2: find `newLocation`
Since the array is a fixed number of 5 (inherited into the `Lesson` contract)
```solidity
bool[5] internal s_booleanArray;
```
I just had to select a number between 0 and 4


**What I don't understand:**
Where/when does the array change? In the `Lesson` contract I see the following array in the constructor:
```solidity
s_booleanArray = [false, false, false, true, false];
```

any guidance would be much appreciated!

**Reply by finessevanes** [reply]
Not sure what code you need to see? I'm looking to learn _where_ the array changes order.
